// ngx-highlightjs/esm2022/lib/highlight-auto.mjs
import { Directive, Input, Output, signal, input, EventEmitter } from '@angular/core';
import { HighlightBase } from './highlight-base';
import * as i0 from "@angular/core";
export class HighlightAuto extends HighlightBase {
    constructor() {
        super(...arguments);
        // Code to highlight
        this.code = input(null, { alias: 'highlightAuto' });
        // Highlighted result
        this.highlightResult = signal(null);
        // Stream that emits when code string is highlighted
        this.highlighted = new EventEmitter();
    }
    async highlightElement(code) {
        const res = await this._hljs.highlightAuto(code, this.languages);
        this.highlightResult.set(res);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: HighlightAuto, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "17.1.0", version: "18.0.1", type: HighlightAuto, isStandalone: true, selector: "[highlightAuto]", inputs: { code: { classPropertyName: "code", publicName: "highlightAuto", isSignal: true, isRequired: false, transformFunction: null }, languages: { classPropertyName: "languages", publicName: "languages", isSignal: false, isRequired: false, transformFunction: null } }, outputs: { highlighted: "highlighted" }, host: { properties: { "class.hljs": "true" } }, providers: [{ provide: HighlightBase, useExisting: HighlightAuto }], usesInheritance: true, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: HighlightAuto, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    selector: '[highlightAuto]',
                    providers: [{ provide: HighlightBase, useExisting: HighlightAuto }],
                    host: {
                        '[class.hljs]': 'true'
                    }
                }]
        }], propDecorators: { languages: [{
                type: Input
            }], highlighted: [{
                type: Output
            }] } });
