// ngx-highlightjs/esm2022/plus/code-from-url.mjs
import { inject, Pipe } from '@angular/core';
import { CodeLoader } from './code-loader';
import { HIGHLIGHT_FILE_LOCATION } from './code-file-location';
import { isUrl } from './gist.model';
import * as i0 from "@angular/core";
export class CodeFromUrlPipe {
    constructor() {
        this._location = inject(HIGHLIGHT_FILE_LOCATION);
        this._loader = inject(CodeLoader);
    }
    transform(url) {
        return this._loader.getCodeFromUrl(isUrl(url) ? url : `${this._location.getPathname()}/${url}`);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: CodeFromUrlPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "18.0.1", ngImport: i0, type: CodeFromUrlPipe, isStandalone: true, name: "codeFromUrl" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: CodeFromUrlPipe, decorators: [{
            type: Pipe,
            args: [{
                    standalone: true,
                    name: 'codeFromUrl'
                }]
        }] });
