// ngx-highlightjs/esm2022/line-numbers/line-numbers.mjs
import { Directive, Input, inject, effect, numberAttribute, booleanAttribute, ElementRef, PLATFORM_ID } from '@angular/core';
import { isPlatformBrowser } from '@angular/common';
import { HighlightJS, HighlightBase, HIGHLIGHT_OPTIONS } from 'ngx-highlightjs';
import * as i0 from "@angular/core";
export class HighlightLineNumbers {
    constructor() {
        this._platform = inject(PLATFORM_ID);
        this.options = inject(HIGHLIGHT_OPTIONS)?.lineNumbersOptions;
        this._hljs = inject(HighlightJS);
        this._highlight = inject(HighlightBase);
        this._nativeElement = inject((ElementRef)).nativeElement;
        this.startFrom = this.options?.startFrom;
        this.singleLine = this.options?.singleLine;
        if (isPlatformBrowser(this._platform)) {
            effect(() => {
                if (this._highlight.highlightResult()) {
                    this.addLineNumbers();
                }
            });
        }
    }
    addLineNumbers() {
        // Clean up line numbers observer
        this.destroyLineNumbersObserver();
        requestAnimationFrame(async () => {
            // Add line numbers
            await this._hljs.lineNumbersBlock(this._nativeElement, {
                startFrom: this.startFrom,
                singleLine: this.singleLine
            });
            // If lines count is 1, the line numbers library will not add numbers
            // Observe changes to add 'hljs-line-numbers' class only when line numbers is added to the code element
            this._lineNumbersObs = new MutationObserver(() => {
                if (this._nativeElement.firstElementChild?.tagName.toUpperCase() === 'TABLE') {
                    this._nativeElement.classList.add('hljs-line-numbers');
                }
                this.destroyLineNumbersObserver();
            });
            this._lineNumbersObs.observe(this._nativeElement, { childList: true });
        });
    }
    destroyLineNumbersObserver() {
        if (this._lineNumbersObs) {
            this._lineNumbersObs.disconnect();
            this._lineNumbersObs = null;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: HighlightLineNumbers, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "16.1.0", version: "18.0.1", type: HighlightLineNumbers, isStandalone: true, selector: "[highlight][lineNumbers], [highlightAuto][lineNumbers]", inputs: { startFrom: ["startFrom", "startFrom", numberAttribute], singleLine: ["singleLine", "singleLine", booleanAttribute] }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "18.0.1", ngImport: i0, type: HighlightLineNumbers, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    selector: '[highlight][lineNumbers], [highlightAuto][lineNumbers]'
                }]
        }], ctorParameters: () => [], propDecorators: { startFrom: [{
                type: Input,
                args: [{ transform: numberAttribute }]
            }], singleLine: [{
                type: Input,
                args: [{ transform: booleanAttribute }]
            }] } });
